tests/core/pyspec/eth2spec/test/altair/epoch_processing/test_process_inactivity_updates.py

```diff
@@ -33,58 +27,232 @@ def test_genesis(spec, state):
     yield from run_process_inactivity_updates(spec, state)
 
 
+@with_altair_and_later
+@spec_state_test
+def test_genesis_random_scores(spec, state):
+    rng = Random(10102)
+    state.inactivity_scores = [rng.randint(0, 100) for _ in state.inactivity_scores]
+    pre_scores = state.inactivity_scores.copy()
+
+    yield from run_process_inactivity_updates(spec, state)
+
+    assert state.inactivity_scores == pre_scores
+
+
 #
 # Genesis epoch processing is skipped
 # Thus all of following tests all go past genesis epoch to test core functionality
 #
 
+def run_inactivity_scores_test(spec, state, participation_fn=None, inactivity_scores_fn=None, rng=Random(10101)):
+    next_epoch_via_block(spec, state)
+    if participation_fn is not None:
+        participation_fn(spec, state, rng=rng)
+    if inactivity_scores_fn is not None:
+        inactivity_scores_fn(spec, state, rng=rng)
+    yield from run_process_inactivity_updates(spec, state)
+
+
 @with_altair_and_later
 @spec_state_test
 def test_all_zero_inactivity_scores_empty_participation(spec, state):
-    next_epoch_via_block(spec, state)
-    state.inactivity_scores = [0] * len(state.validators)
-    yield from run_process_inactivity_updates(spec, state)
+    yield from run_inactivity_scores_test(spec, state, set_empty_participation, zero_inactivity_scores)
+    assert set(state.inactivity_scores) == set([0])
```

Comment:
note: even w/ no participation, the state is not yet in an inactivity leak so any change to the scores is offset by the `RECOVERY` constant